src/stores/lava-simulation-store.ts

```diff
@@ -0,0 +1,98 @@
+import { applySnapshot, types } from "mobx-state-tree";
+import MolassesWorker from "../components/lava-coder/molasses.worker";
+import { AsciiRaster } from "../components/lava-coder/parse-ascii-raster";
+
+// Saving the lava elevations in the MST model is very slow, so we save it separately.
+// But that means that when this is updated, another observable feature (like coveredCells or pulseCount) needs to be
+// updated to trigger reactions. And if we need to save the lava elevations for save/load or some other purpose,
+// this variable should be saved in the MST model.
+export let lavaElevations: number[][] | undefined;
+
+function countCoveredCells(_lavaElevations: number[][]) {
+  let coveredCells = 0;
+  _lavaElevations.forEach(row => {
+    row.forEach(lavaElevation => {
+      if (lavaElevation > 0) {
+        coveredCells++;
+      }
+    });
+  });
+  return coveredCells;
+}
+
+export const LavaSimulationStore = types
+  .model("lavaSimulation", {
+    residual: 5,
+    // Recommended vent
+    ventEasting: 232214,
+    ventNorthing: 2158722,
+    // Vent in the main crater
+    // ventEasting: 227500,
+    // ventNorthing: 2153500,
+    // Vent near the north, quickly hitting a wall and splitting
+    // ventEasting: 237214,
+    // ventNorthing: 2173722,
+    // Vent near the north east corner of the current map, quickly falling into the ocean
+    // ventEasting: 279000,
+    // ventNorthing: 2181000,
+    totalVolume: 200000000,
+    pulseVolume: 100000, // Standard for small eruption
+    pulseCount: 0,
+    lavaElevations: types.array(types.array(types.number)),
```

Comment:
In a comment above, you say that we don't store lava elevations in the model, but here we're storing lava elevations in the model. (?) If we are going to store them in the model, consider making them frozen so MST doesn't make every element of the matrix independently observable.

Reply:
My thought here was that we would want to include the lava elevations in the model for the purposes of saving/loading the simulation. But given that we're not there yet, and we may not even care about saving the state of the lava anyway, I'm just going to remove this.

Reply:
I think it makes sense to leave the model light for now and just add the heavy items if/when we determine a need for them.